src/Compilers/Core/Portable/NativePdbWriter/ISymUnmanagedWriter.cs

```diff
@@ -147,48 +147,107 @@ internal interface IPdbWriter
     [StructLayout(LayoutKind.Explicit)]
     internal struct VariantStructure
     {
-        public VariantStructure(DateTime date) : this() // Need this to avoid errors about the uninteresting union fields.
+        public unsafe VariantStructure(DateTime date) : this() // Need this to avoid errors about the uninteresting union fields.
         {
-            _longValue = date.Ticks;
+            fixed (VariantStructure* thisPtr = &this)
+            {
+                *((long*)((byte*)thisPtr+8)) = date.Ticks;
+            }
+
             _type = (short)VarEnum.VT_DATE;
         }
 
         [FieldOffset(0)]
         private readonly short _type;
 
-        [FieldOffset(8)]
-        private readonly long _longValue;
-
         /// <summary>
         /// This field determines the size of the struct 
         /// (16 bytes on 32-bit platforms, 24 bytes on 64-bit platforms).
         /// </summary>
         [FieldOffset(8)]
-        private readonly VariantPadding _padding;
+        private readonly VariantData _data;
 
-        // Fields below this point are only used to make inspecting this struct in the debugger easier.
+        // The following properties are only used to make inspecting the native memory in the debugger easier.
```

Comment:
>// The following properties are only used to make inspecting the native memory in the debugger easier. [](start = 8, length = 102)

If that is so, I'd just delete them. I don't think it's worth having a bunch of unsafe code.